src/SWP/Bundle/ContentBundle/Resources/config/app/config.yml

```diff
@@ -6,6 +6,7 @@ parameters:
     env(FS_AWS_S3_REGION): ''
     env(FS_AWS_S3_KEY): ''
     env(FS_AWS_S3_SECRET): ''
+    env(FS_AWS_S3_ENDPOINT): null
```

Comment:
@kottkrig please set here default aws s3 endpoint. 

```Endpoints must be full URIs and include a scheme and host ```

Reply:
It can be just an empty string according to this: https://github.com/guzzle/psr7/blob/master/src/Uri.php#L77

@kottkrig I saw you set it to `env(FS_AWS_S3_ENDPOINT)= ''` before then changed to `env(FS_AWS_S3_ENDPOINT): null`, the empty string didn't work?

Reply:
I believe that I got the same error with an empty string but I can try it again.

Reply:
Yes. The same error with an empty string:

> Endpoints must be full URIs and include a scheme and host

Reply:
yep, to solve this problem, you can wrap the instantiation of `Aws\S3\S3Client` in a new class and pass the arguments (`version`, `region` etc.) into it, so if the `env(FS_AWS_S3_ENDPOINT)= ''` env var is an empty string then `endpoint` argument should not be set in the `Aws\S3\S3Client` at all.
You can try this https://symfony.com/doc/4.3/service_container/factories.html

Reply:
Thank you. I will have a go at it in a couple of days.

Reply:
Guys, why not setting default aws endpoint as default value?

Reply:
it's because different regions can have different endpoints https://docs.aws.amazon.com/general/latest/gr/s3.html and the default value of `endpoint` must be valid URL or not set at all in order to fall back to a default endpoint in the PHP client. It's should be optional to set the value of `endpoint`

Reply:
@takeit - then yes - we need some special logic in wrapper set before the client.